src/ImageSharp.Web/Providers/FileProviderImageProvider.cs

```diff
@@ -29,21 +29,26 @@ public class FileProviderImageProvider : IImageProvider
         /// Initializes a new instance of the <see cref="FileProviderImageProvider"/> class.
         /// </summary>
         /// <param name="fileProvider">The file provider.</param>
+        /// <param name="processingBehavior">The processing behavior.</param>
         /// <param name="formatUtilities">Contains various format helper methods based on the current configuration.</param>
-        public FileProviderImageProvider(IFileProvider fileProvider, FormatUtilities formatUtilities)
+        protected FileProviderImageProvider(IFileProvider fileProvider, ProcessingBehavior processingBehavior, FormatUtilities formatUtilities)
         {
-            this.fileProvider = fileProvider ?? throw new ArgumentNullException(nameof(fileProvider));
-            this.formatUtilities = formatUtilities ?? throw new ArgumentNullException(nameof(formatUtilities));
+            Guard.NotNull(fileProvider, nameof(fileProvider));
+            Guard.NotNull(formatUtilities, nameof(formatUtilities));
+
+            this.fileProvider = fileProvider;
+            this.formatUtilities = formatUtilities;
+            this.ProcessingBehavior = processingBehavior;
         }
 
         /// <inheritdoc/>
-        public ProcessingBehavior ProcessingBehavior { get; protected set; } = ProcessingBehavior.CommandOnly;
+        public ProcessingBehavior ProcessingBehavior { get; }
```

Comment:
Or remove the `processingBehavior` parameter from the constructor and make the property virtual again (aligning it with the `Match` property):

```suggestion
        public virtual ProcessingBehavior ProcessingBehavior { get; } = ProcessingBehavior.CommandOnly;
```

The `PhysicalFileSystemProvider` implmentation can then override this property to be able to set the value in the constructor and `WebRootImageProvider` can simply use the default value (without having to specify it when calling the base constructor).

My reasoning for this is that base/abstract class shouldn't be influenced by derived classes, so you should add constructor parameters because one single derived class want to set this property.

Reply:
>My reasoning for this is that base/abstract class shouldn't be influenced by derived classes, so you should add constructor parameters because one single derived class want to set this property.

This doesn't make any sense. Base class implementations absolutely are influenced by the parameters passed. 

I'm adding the constructor parameter to ensure that the API shape matches the interface. `ProcessingBehavior` is a readonly required field, therefor it should be passed via the constructor.